integration/src/main/java/com/aws/greengrass/mqttbroker/ClientDeviceAuthorizer.java

```diff
@@ -86,7 +77,12 @@ public boolean checkValid(String clientId, String username, byte[] password) {
     @Override
     public boolean canWrite(Topic topic, String user, String client) {
         String resource = "mqtt:topic:" + topic;
-        boolean canPerform = canDevicePerform(getSessionForClient(client, user), "mqtt:publish", resource);
+        boolean canPerform = false;
+        try {
+            canPerform = canDevicePerform(getSessionForClient(client, user), "mqtt:publish", resource);
+        } catch (AuthenticationException e) {
+            LOG.atWarn().kv(CLIENT_ID, client).log("Unable to re-authenticate client.");
+        }
         LOG.atDebug().kv("topic", topic).kv("isAllowed", canPerform).kv(CLIENT_ID, client)
             .log("MQTT publish request");
         return canPerform;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 79:88, 93:102.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.